### ISSUE TITLE
Dns issues with node-local

### ISSUE THREAD

**Issue by donkeyx** [rancher/rke#2372]
**RKE version:**
`rke version v1.0.14`

**Docker version: (`docker version`,`docker info` preferred)**
```shell
Client: Docker Engine - Community
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 17:02:55 2020
 OS/Arch:           linux/amd64
 Experimental:      false
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```shell
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
`proxmox vm`

**cluster.yml file:**
https://gist.github.com/donkeyx/efa8ef2297df801c00948ebf44656dad

**Steps to Reproduce:**
Sandup cluster using above config and dns requests will get timeouts, which can be seen in node-local logs. 

Rke has been working well with node-local until the last update. After this, my pods are crashing with timeouts. I read that I should be adding `iptables -P FORWARD ACCEPT` to allow core-dns to work, which seems to have gotten it mostly working. But, I am still getting lots of timeouts. 

It actually looks like its appending kube dns onto normal dns requests. But, also failing with normal requests within the cluster, that's probably not the issue.

**Results:**
errors in node-local which are the timeouts the application is getting. 

```shell
node-local-dns-9bh5d node-cache  [ERROR] plugin/errors: 2 postgres-96-01.internal.mydomain.com.svc.cluster.local. A: read tcp 10.172.10.40:45042->10.43.250.171:53: i/o timeout
node-local-dns-9bh5d node-cache  [ERROR] plugin/errors: 2 dev-stack-05.internal.sine.co.cluster.local. A: read tcp 10.172.10.40:45312->10.43.250.171:53: i/o timeout
node-local-dns-9bh5d node-cache  [ERROR] plugin/errors: 2 mock-sns-sqs-svc.dev-stack-08.svc.cluster.local.dev-stack-08.svc.cluster.local. A: read tcp 10.172.10.40:45556->10.43.250.171:53: i/o timeout
node-local-dns-vhv4k node-cache  [ERROR] plugin/errors: 2 zipkin.istio-system.dev-stack-07.svc.cluster.local. A: read tcp 10.172.10.30:46266->10.43.250.171:53: i/o timeout
node-local-dns-9bh5d node-cache  [ERROR] plugin/errors: 2 mock-sns-sqs-svc.dev-stack-07.svc.cluster.local. A: read tcp 10.172.10.40:46318->10.43.250.171:53: i/o timeout
node-local-dns-9bh5d node-cache  [ERROR] plugin/errors: 2 mock-sns-sqs-svc.dev-stack-07.svc.cluster.local. A: read tcp 10.172.10.40:46852->10.43.250.171:53: i/o timeout
```

I am not sure how to solve, should i just remove nodelocal by commenting and `rke up` again?

**Comment by superseb** [comment]
Does `v1.0.13` work fine? What Kubernetes version are you using/configuring, does it work with one and not the other? Are we talking upgrades (if so, from what version) or clean install?

**Comment by donkeyx** [comment]
I have it working now by rolling back to rke 1.2.3 which has resolved the dns issue, but I still have the forwarding rule in place which may or may not be needed now?

This started with an `rke up` using v1.0.14, and no other changes, which basically put istiod and coredns in a state that could not start. After I added the forward rule, they came up ok, but then dns was very flake, worked a few times then failed the next few with node local not resolving the requests.  

Settings with this broken state: 

```shell
▶ rke version
INFO[0000] Running RKE version: v1.0.14
FATA[0000] Failed to create Kubernetes Client: stat ./kube_config_cluster.yml: no such file or directory


▶ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-12T01:09:16Z", GoVersion:"go1.15.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.14", GitCommit:"f238f5142728be4033c37aa0ad69bf806090beae", GitTreeState:"clean", BuildDate:"2020-11-11T13:03:54Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```


Settings when I had it working again: 
```shell
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-12T01:09:16Z", GoVersion:"go1.15.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:09:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by superseb** [comment]
Not really sure if I follow, 1.2.3 is the latest release for 1.2.x and there is no path from 1.2 to 1.x that is lower than 1.2. The next available upgrade is 1.2.4 when it releases. Please confirm what version was in use before and after, the output also shows the server version is lower.

**Comment by donkeyx** [comment]
My bad @superseb, I just pulled the latest release from github thinking it was just a numbering change.
